### PR TITLE
Doc: Add groups to identity concepts

### PR DESCRIPTION
**Pull request by vishalnayak**

### CODE REVIEW

File: website/source/docs/enterprise/identity/index.html.md [hunk 1]
```diff
@@ -39,45 +39,77 @@ performing actions allowed by the policies on the entity as well.
...
+Entities can be direct members of groups, in which case they inherit the
+policies of the groups they belong to. Entities can also be indirect members of
+groups. For example, if a GroupA has GroupB as subgroup, then members of GroupB
+are indirect members of GroupA. Hence, the members of GroupB will have access
+to policies on both GroupA and GroupB.
+
+### External vs Internal Groups
+
+By default, the groups created in identity store are called the internal
+groups.  The membership management of these groups should be carried out
```
Comment (calvn) [inline website/source/docs/enterprise/identity/index.html.md]: Double spaces in `.  The membership management`

File: website/source/docs/enterprise/identity/index.html.md [hunk 2]
```diff
@@ -39,45 +39,77 @@ performing actions allowed by the policies on the entity as well.
...
+manually.  A group can also be created as an external group. In this case, the
```
Comment (calvn) [inline website/source/docs/enterprise/identity/index.html.md]: Double spaces in `.  A group can also`

File: website/source/docs/enterprise/identity/index.html.md [hunk 3]
```diff
@@ -18,16 +18,16 @@ the tokens used in Vault.
...
+made up of zero or more aliases.
 
 ### Entity Management
 
 Entities in Vault **do not** automatically pull identity information from
 anywhere. It needs to be explicitly managed by operators. This way, it is
 flexible in terms of administratively controlling the number of entities to be
-pulled in and pulled out of Vault, and in some sense Vault will serve as a
-_cache_ of identities and not as the _source_ of identities.
+pulled-in and pulled-out of Vault. In some sense, Vault will serve as a _cache_
```
[the remark this answers is not in the document]
Comment (vishalnayak) [reply website/source/docs/enterprise/identity/index.html.md]: Done

File: website/source/docs/enterprise/identity/index.html.md (hunk 2 above)
Comment (vishalnayak) [reply website/source/docs/enterprise/identity/index.html.md]: Fixed.

**Pull request merged**